packages/app-desktop/gui/Dialog.tsx

```diff
@@ -63,23 +63,70 @@ const Dialog: React.FC<Props> = props => {
 	</div>;
 };
 
+// We keep track of the mouse events to allow the action to be cancellable on the mouseup
+// If dialogElement is the source of the mouse event it means
+// that the user clicked in the dimmed background and not in the content of the dialog
+const useClickedOutsideContent = (dialogElement: HTMLDialogElement|null) => {
+	const mouseDownOutsideContent = useRef(false);
+	mouseDownOutsideContent.current	= false;
+	const [clickedOutsideContent, setClickedOutsideContent] = useState(false);
+
+	useEffect(() => {
+		if (!dialogElement) return () => {};
+
+		const mouseDownListener = (event: MouseEvent) => {
+			if (event.target === dialogElement) {
+				mouseDownOutsideContent.current = true;
+			} else {
+				mouseDownOutsideContent.current = false;
+			}
+		};
+		const mouseUpListener = (event: MouseEvent) => {
+			if (!mouseDownOutsideContent) return;
```

Comment:
```suggestion
```

**Optional**: Since `mouseDownOutsideContent` is a `ref`, I think `!mouseDownOutsideContent` should always be `false`.